### PR TITLE
docs: update dependency list in NOTICE.md file

### PR DESCRIPTION
**Pull request by mukta2022**
## What this PR changes/adds

Here, update NOTICE.md file  by using Eclipse Dash Tool (https://github.com/eclipse/dash-licenses)

## Why it does that

update all relevant dependency list used in Connector repo

## Checklist

- [ ] added appropriate tests?
- [ ] performed checkstyle check locally?
- [ ] added/updated relevant documentation?
- [ ] assigned appropriate label? 
- [ ] formatted title correctly?

### CODE REVIEW

File: NOTICE.md [hunk 1]
```diff
@@ -2874,167 +2673,151 @@ SLF4J API Module
...
-
-Logback Classic Module
-* Maven Artifact: ch.qos.logback/logback-classic/1.2.10
-* License: EPL-1.0
-
-Logback Core Module
-* Maven Artifact: ch.qos.logback/logback-core/1.2.10
-* License: EPL-1.0
-
-## Cryptography
```
Comment (juliapampus) [inline NOTICE.md:3038]: Please don't delete this section

File: NOTICE.md [hunk 2]
```diff
@@ -26,36 +26,36 @@ The project maintains the following source code repositoriy:
 
 * https://github.com/eclipse-edc/Connector
 
-## Third-party Content (Overarching All Modules)
+## Third Party Content
```
Comment (alexandrudanciu) [inline NOTICE.md:29]: The hyphen should be kept as third-party acts as adjective to content. I agree to use a capital letter for "Party" though.

File: NOTICE.md [hunk 3]
```diff
@@ -531,71 +453,81 @@ Jackson-datatype-Joda
...
-* Maven Artifact: com.fasterxml.jackson.datatype/jackson-datatype-jsr310/2.12.5
+* Maven Artifact: com.fasterxml.jackson.datatype/jackson-datatype-jsr310/2.14.2
 * License: Apache-2.0
 
-Jackson datatype: JSR310
-* Maven Artifact: com.fasterxml.jackson.datatype/jackson-datatype-jsr310/2.13.1
-* License: Apache-2.0
+Jackson Jakarta-RS: base
+* Maven Artifact: com.fasterxml.jackson.jakarta.rs/jackson-jakarta-rs-base/2.14.2
+* License:
```
Comment (alexandrudanciu) [inline NOTICE.md:473]: Missing license

File: NOTICE.md [hunk 4]
```diff
@@ -531,71 +453,81 @@ Jackson-datatype-Joda
...
 
-Jackson datatype: JSR310
-* Maven Artifact: com.fasterxml.jackson.datatype/jackson-datatype-jsr310/2.13.3
-* License: Apache-2.0
+Jackson Jakarta-RS: JSON
+* Maven Artifact: com.fasterxml.jackson.jakarta.rs/jackson-jakarta-rs-json-provider/2.13.3
+* License:
```
Comment (alexandrudanciu) [inline NOTICE.md:477]: Missing license

File: NOTICE.md [hunk 5]
```diff
@@ -531,71 +453,81 @@ Jackson-datatype-Joda
...
+* Maven Artifact: com.fasterxml.jackson/jackson-bom/2.13.3
 * License: Apache-2.0
 
 Jackson BOM
 * Project: https://github.com/FasterXML/jackson-bom
 * Source: scm:git:git@github.com:FasterXML/jackson-bom.git
-* Maven Artifact: com.fasterxml.jackson/jackson-bom/2.13.3
-* License: Apache-2.0
+* Maven Artifact: com.fasterxml.jackson/jackson-bom/2.14.0
+* License:
```
Comment (alexandrudanciu) [inline NOTICE.md:513]: Missing license

File: NOTICE.md [hunk 6]
```diff
@@ -626,13 +552,13 @@ jackson-coreutils-equivalence
...
 * Maven Artifact: com.github.java-json-tools/jackson-coreutils-equivalence/1.0
-* License: Apache-2.0
+* License: LGPL-3.0
 
 jackson-coreutils
 * Project: https://github.com/java-json-tools/jackson-coreutils
 * Source: git@github.com:java-json-tools/jackson-coreutils.git
 * Maven Artifact: com.github.java-json-tools/jackson-coreutils/2.0
-* License: Apache-2.0
+* License:
```
Comment (alexandrudanciu) [inline NOTICE.md:561]: Missing license

File: NOTICE.md [hunk 7]
```diff
@@ -644,7 +570,7 @@ json-schema-core
 * Project: https://github.com/java-json-tools/json-schema-core
 * Source: git@github.com:java-json-tools/json-schema-core.git
 * Maven Artifact: com.github.java-json-tools/json-schema-core/1.2.14
-* License: Apache-2.0
+* License:
```
Comment (alexandrudanciu) [inline NOTICE.md:573]: Missing license

File: NOTICE.md [hunk 8]
```diff
@@ -656,26 +582,190 @@ msg-simple
...
+* Project: https://github.com/googleapis/java-iam/proto-google-iam-v1
+* Source: scm:git:git@github.com:googleapis/java-iam.git/proto-google-iam-v1
+* Maven Artifact: com.google.api.grpc/proto-google-iam-v1/1.5.2
+* License: Apache-2.0
+
+proto-google-iam-v1
+* Project: https://github.com/googleapis/gapic-generator-java
+* Source: scm:git:git@github.com:googleapis/gapic-generator-java.git
+* Maven Artifact: com.google.api.grpc/proto-google-iam-v1/1.9.2
+* License:
```
Comment (alexandrudanciu) [inline NOTICE.md:655]: Missing license

File: NOTICE.md [hunk 9]
```diff
@@ -656,26 +582,190 @@ msg-simple
...
+* Project: https://github.com/googleapis/google-cloud-java
+* Source: scm:git:git@github.com:googleapis/google-cloud-java.git
+* Maven Artifact: com.google.cloud/google-cloud-core-grpc/2.12.0
+* License:
+
+Google Cloud Core HTTP
+* Project: https://github.com/googleapis/google-cloud-java
+* Source: scm:git:git@github.com:googleapis/google-cloud-java.git
+* Maven Artifact: com.google.cloud/google-cloud-core-http/2.12.0
+* License:
```
Comment (alexandrudanciu) [inline NOTICE.md:743]: Missing license

File: NOTICE.md [hunk 10]
```diff
@@ -656,26 +582,190 @@ msg-simple
...
+* Project: https://github.com/googleapis/google-cloud-java
+* Source: scm:git:git@github.com:googleapis/google-cloud-java.git
+* Maven Artifact: com.google.cloud/google-cloud-core/2.12.0
+* License:
+
+Google Cloud IAM Service Account Credentials
+* Project: https://github.com/googleapis/java-iamcredentials
+* Source: scm:git:git@github.com:googleapis/java-iamcredentials.git/google-cloud-iamcredentials
+* Maven Artifact: com.google.cloud/google-cloud-iamcredentials/2.3.5
+* License:
```
Comment (alexandrudanciu) [inline NOTICE.md:755]: Missing license

File: NOTICE.md [hunk 11]
```diff
@@ -656,26 +582,190 @@ msg-simple
...
+* Project: https://github.com/googleapis/java-storage
+* Source: scm:git:git@github.com:googleapis/java-storage.git/google-cloud-storage
+* Maven Artifact: com.google.cloud/google-cloud-storage/2.20.1
+* License:
+
+Google IAM Admin API
+* Project: https://github.com/googleapis/java-iam-admin
+* Source: scm:git:git@github.com:googleapis/java-iam-admin.git/google-iam-admin
+* Maven Artifact: com.google.cloud/google-iam-admin/1.2.5
+* License:
```
Comment (alexandrudanciu) [inline NOTICE.md:767]: Missing license

File: NOTICE.md [hunk 12]
```diff
@@ -732,22 +835,75 @@ Guava: Google Core Libraries for Java
...
+* Project: http://github.com/google/re2j
+* Maven Artifact: com.google.re2j/re2j/1.6
+* License: BSD-3-Clause
+
 H2 Database Engine
 * Project: https://h2database.com
 * Source: scm:git:https://github.com/h2database/h2database
 * Maven Artifact: com.h2database/h2/2.1.210
-* License: EPL-1.0
+* License: (EPL-1.0
```
Comment (alexandrudanciu) [inline NOTICE.md:906]:
```suggestion
* License: EPL-1.0
```

File: NOTICE.md [hunk 13]
```diff
@@ -1048,24 +1154,20 @@ Apache Commons Logging
 * Maven Artifact: commons-logging/commons-logging/1.2
 * License: Apache-2.0
 
+Failsafe OkHttp
+* Maven Artifact: dev.failsafe/failsafe-okhttp/3.2.4
+* License:
```
Comment (alexandrudanciu) [inline NOTICE.md:1159]: Missing license

File: NOTICE.md [hunk 14]
```diff
@@ -2002,7 +1837,7 @@ javax.ws.rs-api
 * Project: http://jax-rs-spec.java.net
 * Source: scm:git:https://github.com/jax-rs/api.git
 * Maven Artifact: javax.ws.rs/javax.ws.rs-api/2.1
-* License: CDDL-1.1
+* License:
```
Comment (alexandrudanciu) [inline NOTICE.md:1840]: Missing license

File: NOTICE.md [hunk 15]
```diff
@@ -2636,28 +2524,28 @@ LatencyUtils
...
-* Maven Artifact: org.mock-server/mockserver-client-java/5.14.0
+* Maven Artifact: org.mock-server/mockserver-client-java/5.15.0
 * License: Apache-2.0
 
 MockServer Core
 * Project: https://www.mock-server.com
-* Maven Artifact: org.mock-server/mockserver-core/5.14.0
-* License: Apache-2.0
+* Maven Artifact: org.mock-server/mockserver-core/5.15.0
+* License:
```
Comment (alexandrudanciu) [inline NOTICE.md:2533]: Missing license

File: NOTICE.md [hunk 16]
```diff
@@ -2636,28 +2524,28 @@ LatencyUtils
...
-* Maven Artifact: org.mock-server/mockserver-netty/5.14.0
+* Maven Artifact: org.mock-server/mockserver-netty/5.15.0
 * License: Apache-2.0
 
 mockito-core
 * Project: https://github.com/mockito/mockito
-* Maven Artifact: org.mockito/mockito-core/4.2.0
-* License: MIT
+* Maven Artifact: org.mockito/mockito-core/5.2.0
+* License:
```
Comment (alexandrudanciu) [inline NOTICE.md:2543]: Missing license

File: NOTICE.md [hunk 17]
```diff
@@ -2636,28 +2524,28 @@ LatencyUtils
...
 
 mockito-inline
 * Project: https://github.com/mockito/mockito
-* Maven Artifact: org.mockito/mockito-inline/4.2.0
-* License: MIT
+* Maven Artifact: org.mockito/mockito-inline/5.2.0
+* License:
```
Comment (alexandrudanciu) [inline NOTICE.md:2548]: Missing license

File: NOTICE.md [hunk 18]
```diff
@@ -2874,167 +2681,153 @@ SLF4J API Module
...
-* Maven Artifact: software.amazon.awssdk/arns/2.16.60
+* Maven Artifact: software.amazon.awssdk/arns/2.19.26
 * License: Apache-2.0
 
 AWS Java SDK :: Auth
 * Project: https://aws.amazon.com/sdkforjava
-* Maven Artifact: software.amazon.awssdk/auth/2.16.60
-* License: Apache-2.0
+* Maven Artifact: software.amazon.awssdk/auth/2.19.26
+* License:
```
Comment (alexandrudanciu) [inline NOTICE.md:2735]: Missing license

File: NOTICE.md [hunk 19]
```diff
@@ -815,63 +971,57 @@ Java Client Runtime for AutoRest
...
+* Maven Artifact: com.nimbusds/oauth2-oidc-sdk/9.35
 * License: Apache-2.0
 
 checkstyle
 * Project: https://checkstyle.org/
 * Source: scm:git:git@github.com:checkstyle/checkstyle.git
-* Maven Artifact: com.puppycrawl.tools/checkstyle/9.0
-* License: Apache-2.0
+* Maven Artifact: com.puppycrawl.tools/checkstyle/10.0
+* License: LGPL-2.1+
```
Comment (alexandrudanciu) [inline NOTICE.md:1017]: We would probably not use this license as foundation. The project probably offers multiple alternative licenses.

File: NOTICE.md [hunk 20]
```diff
@@ -1974,17 +1803,23 @@ Jakarta XML Binding API
...
+* Source: scm:git:https://github.com/javaee/javax.annotation.git
+* Maven Artifact: javax.annotation/javax.annotation-api/1.3.2
+* License: CDDL-1.1
+
 Java Servlet API
 * Project: http://servlet-spec.java.net
 * Source: scm:svn:https://svn.java.net/svn/glassfish~svn/tags/javax.servlet-api-3.1.0
 * Maven Artifact: javax.servlet/javax.servlet-api/3.1.0
-* License: CDDL-1.1
+* License: (CDDL-1.1
```
Comment (alexandrudanciu) [inline NOTICE.md:1816]:
```suggestion
* License: CDDL-1.1
```

File: NOTICE.md [hunk 21]
```diff
@@ -1974,17 +1803,23 @@ Jakarta XML Binding API
...
 
 Java Servlet API
 * Project: https://javaee.github.io/servlet-spec/
 * Source: scm:git:https://github.com/javaee/servlet-spec.git
 * Maven Artifact: javax.servlet/javax.servlet-api/4.0.1
-* License: CDDL-1.1
+* License: (CDDL-1.1
```
Comment (alexandrudanciu) [inline NOTICE.md:1822]:
```suggestion
* License: CDDL-1.1
```

**Review by alexandrudanciu — COMMENTED** [review]
Proposed some minor improvements.

**Review by alexandrudanciu — COMMENTED** [review]
Several review comments were ignored. Provided missing values as code changes.

**Pull request closed without merging**